transpileModule fails when declarationMap is specified

### 🔎 Search Terms

transpileModule declarationMap

### 🕗 Version & Regression Information

- This changed between versions 5.4 and 5.5

### ⏯ Playground Link

_No response_

### 💻 Code

_No response_

### 🙁 Actual behavior

When calling transpileModule with options that specify both `declaration` and `declarationMap`, `transpileModule` will force to false `declaration` but will leave `declarationMap` as is resulting in the error: `error TS5069: Option 'declarationMap' cannot be specified without specifying option 'declaration' or option 'composite'.`

### 🙂 Expected behavior

`transpileModule` should set both `declaration` and `declarationMap` to false (or neither) 

### Additional information about the issue

_No response_

--- a/src/services/transpile.ts
+++ b/src/services/transpile.ts
@@ -149,6 +149,7 @@ function transpileWorker(input: string, transpileOptions: TranspileOptions, decl
     }
     else {
         options.declaration = false;
+        options.declarationMap = false;
     }
 
     const newLine = getNewLineCharacter(options);

--- a/src/testRunner/unittests/services/transpile.ts
+++ b/src/testRunner/unittests/services/transpile.ts
@@ -294,6 +294,11 @@ var x = 0;`,
         testVerbatimModuleSyntax: true,
     });
 
+    transpilesCorrectly("Supports setting 'declarationMap'", "x;", {
+        options: { compilerOptions: { declarationMap: true }, fileName: "input.js", reportDiagnostics: true },
+        testVerbatimModuleSyntax: true,
+    });
+
     transpilesCorrectly("Supports setting 'declarationDir'", "x;", {
         options: { compilerOptions: { declarationDir: "out/declarations" }, fileName: "input.js", reportDiagnostics: true },
         testVerbatimModuleSyntax: true,

--- a/tests/baselines/reference/transpile/Supports setting declarationMap (verbatimModuleSyntax=true).js
+++ b/tests/baselines/reference/transpile/Supports setting declarationMap (verbatimModuleSyntax=true).js
@@ -1,0 +1,2 @@
+x;
+//# sourceMappingURL=input.js.map

--- a/tests/baselines/reference/transpile/Supports setting declarationMap (verbatimModuleSyntax=true).oldTranspile.js
+++ b/tests/baselines/reference/transpile/Supports setting declarationMap (verbatimModuleSyntax=true).oldTranspile.js
@@ -1,0 +1,2 @@
+x;
+//# sourceMappingURL=input.js.map

--- a/tests/baselines/reference/transpile/Supports setting declarationMap.js
+++ b/tests/baselines/reference/transpile/Supports setting declarationMap.js
@@ -1,0 +1,2 @@
+x;
+//# sourceMappingURL=input.js.map

--- a/tests/baselines/reference/transpile/Supports setting declarationMap.oldTranspile.js
+++ b/tests/baselines/reference/transpile/Supports setting declarationMap.oldTranspile.js
@@ -1,0 +1,2 @@
+x;
+//# sourceMappingURL=input.js.map

